modules/EVSE/Evse15118D20/charger/ISO15118_chargerImpl.hpp

```diff
@@ -92,10 +95,13 @@ class ISO15118_chargerImpl : public ISO15118_chargerImplBase {
     iso15118::d20::EvseSetupConfig setup_config;
     std::bitset<NUMBER_OF_SETUP_STEPS> setup_steps_done{0};
 
-    std::vector<iso15118::d20::SupportedVASs> supported_vas_services_per_provider;
+    std::vector<types::iso15118_vas::OfferedServices> supported_vas_services_per_provider;
+    std::map<uint16_t, std::pair<std::size_t, std::reference_wrapper<types::iso15118_vas::OfferedService>>>
+        vas_service_id_to_provider_map;
```

Comment:
I don't quite understand why `supported_vas_services_per_provider` has been changed here. D20 only needs the ServiceIDs from the updated OfferedServices type. So why can't we continue to use the old vector?

I also don't understand the need for the new `vas_service_id_to_provider_map` variable.

Reply:
The original idea was to store the complete offered service instance in case it might be used later. But you're right, this does not make sense for D20 since the data is not used at all. I'll revert to a minimal adaption.